pkgs/os-specific/linux/iwd/default.nix

```
@@ -56,6 +57,14 @@ stdenv.mkDerivation rec {
     "--with-systemd-networkdir=${placeholder "out"}/lib/systemd/network/"
   ];
 
+  patches = [
+    # Fix failure in test-eapol. Remove when bumping to 1.18
+    (fetchpatch {
+      url = "https://git.kernel.org/pub/scm/network/wireless/iwd.git/patch/?id=ed10b00afa3f4c087b46d7ba0b60a47bd05d8b39";
+      sha256 = "0a6lz0wig5bsjvyx9f7r7amwi1ykapdqm1339ixa4356824mmjz8";
```

Do you have any ideas with the hash mismatch? This is what I got:

```suggestion
      sha256 = "0n8ixrbfh428ajncakcb9kd2n4fw82kw9sfskn1d9ny0lrg39nvg";
```

---

No clue tbh? I keep getting the same hash:
```
 nix-prefetch-url "https://git.kernel.org/pub/scm/network/wireless/iwd.git/patch/?id=ed10b00afa3f4c087b46d7ba0b60a47bd05d8b39"
[0.0 MiB DL]
path is '/nix/store/p2fxrjhj8rja0d05fvjfvglhx16gj14a-?id=ed10b00afa3f4c087b46d7ba0b60a47bd05d8b39'
0a6lz0wig5bsjvyx9f7r7amwi1ykapdqm1339ixa4356824mmjz8
```

---

I am not sure now as I can also get your hash with `nix-prefetch-url`, can you build this with the hash you provide?

Not sure if this is related https://github.com/NixOS/nixpkgs/issues/7229, I am getting my hash using [nix-prefetch](https://github.com/msteen/nix-prefetch):

```
nix-shell -p nix-prefetch --run "nix-prefetch fetchpatch --url https://git.kernel.org/pub/scm/network/wireless/iwd.git/patch/?id=ed10b00afa3f4c087b46d7ba0b60a47bd05d8b39"
```

---

Yeah, it builds fine, as noted in the PR. It also builds fine with your hash though. I'm not really sure what to do, if anything, in this case?

---

`fetchpatch` cleans the diff so the hash returned by `nix-prefetch-url` cannot be used (unless the downloaded file would not change by cleaning).

---

Ack, that makes sense. I fixed it up to use the other hash then (I'm guessing it wasn't a problem when building for me since I had prefetched the patch already at that point).